examples/bin/start-druid

```diff
@@ -0,0 +1,548 @@
+    TASK_MEM_TYPE_HIGH: ["-Xms1g", "-Xmx1g", "-XX:MaxDirectMemorySize=1g"]
+}
+
+BROKER = "broker"
+ROUTER = "router"
+COORDINATOR = "coordinator-overlord"
+HISTORICAL = "historical"
+MIDDLE_MANAGER = "middleManager"
+TASKS = "tasks"
+
+DEFAULT_SERVICES = [
+    BROKER,
+    ROUTER,
+    COORDINATOR,
+    HISTORICAL,
+    MIDDLE_MANAGER
+]
+
+SERVICE_MEMORY_RATIO = {
+    MIDDLE_MANAGER: 1,
+    ROUTER: 2,
+    COORDINATOR: 30,
+    BROKER: 46,
+    HISTORICAL: 80,
+    TASKS: 30
+}
+
+MINIMUM_MEMORY_MB = {
+    MIDDLE_MANAGER: 64,
+    ROUTER: 128,
+    TASKS: 1024,
+    BROKER: 900,
+    COORDINATOR: 256,
+    HISTORICAL: 900
+}
+
+HEAP_TO_TOTAL_MEM_RATIO = {
+    MIDDLE_MANAGER: 1,
+    ROUTER: 1,
+    COORDINATOR: 1,
+    BROKER: 0.60,
+    HISTORICAL: 0.40,
+    TASKS: 0.50
+}
+
+LOGGING_ENABLED = False
+
+
+def print_if_verbose(message):
+    if LOGGING_ENABLED:
+        print(message)
+
+
+def configure_parser():
+    parser = argparse.ArgumentParser(
+        prog='start-druid',
+        formatter_class=argparse.RawTextHelpFormatter,
+        epilog=
+        """
+sample usage:
+    start-druid
+            Start up all the services (including zk).
+    start-druid -m=100g
+            Start up all the services (including zk)
+            using a total memory of 100GB.
+    start-druid -m=100g --compute
+            Compute memory distribution and validate arguments.
+    start-druid -m=100g -s=broker,router
+            Starts a broker and a router, using a total memory of 100GB.
+    start-druid -m=100g --s=broker,router \\
+    -c=conf/druid/single-server/custom
+            Starts a broker and a router, using a total memory of 100GB.
+            Reads configs for each service (jvm.config, runtime.properties)
+            from respective folders inside the given root config path.
+    start-druid -s=broker,router \\
+    -c=conf/druid/single-server/custom
+            Starts a broker and a router service, reading service configs
+            from the given root directory. Calculates memory requirements for
+            each service, if required, using upto 80% of the total system memory.
+    start-druid -m=100g \\
+    -s=broker,router \\
+    -c=conf/druid/single-server/custom \\
+    --zk
+            Starts broker, router and zookeeper.
+            zookeeper config is read from conf/zk.
+"""
+    )
+    parser.add_argument('--memory', '-m', type=str, required=False,
+                        help='Total memory for all processes (services and tasks, if any). \n'
+                             'This parameter is ignored if each service already has a jvm.config \n'
+                             'in the given conf directory. e.g. 500m, 4g, 6g\n')
+    parser.add_argument('--services', '-s', type=str, required=False,
+                        help='List of services to be started, subset of \n'
+                             '{broker, router, middleManager, historical, coordinator-overlord}. \n'
```

Comment:
what about `indexer`?

Reply:
can we run `coordinator` and `overlord` separately or are we officially removing that option?

Reply:
`indexer` is not supported yet since we had started based on the `single-server` profiles. We can add support for this in a follow up PR. 
As for `coordinator/overlord`, both `cluster` and `single-server` confs seem to be launching them as one, but we should support launching them separately, maybe in a later PR.

Reply:
> what about indexer?

Running `indexer` isn't supported at the moment, reason being that I was targeting compatibility with the single-server launch scripts and they don't run indexer. Let me see if I can support running indexer. 

> can we run coordinator and overlord separately or are we officially removing that option?

I haven't given it a try, reason being same as above. I will check and get back on this.

Reply:
IMO it's ok to omit support for these items in this initial version, since the initial target is to replicate what the prewritten configs do. We can add these items later if needed.

Reply:
I guess I was asking in the context of my flow right now for my local Druid is:

- build Druid
- apply some changes to the micro-quickstart configs (switch from MM to indexer for my local testing)
- run `./bin/start-micro-quickstart`

I was wondering how I could update that workflow once `./bin/start-druid` becomes a thing. Can I get `./bin/start-druid` to write out the configs for me? I want to use the awesome memory tuning features but I also want to use the indexer locally. What do?

Reply:
> I was wondering how I could update that workflow once ./bin/start-druid becomes a thing. Can I get ./bin/start-druid to write out the configs for me? I want to use the awesome memory tuning features but I also want to use the indexer locally. What do?

This script doesn't recognise `indexer` service as of now. So I think we will have to use one of the hard-coded profiles.